core/amber/src/main/scala/edu/uci/ics/amber/engine/common/storage/partition/PartitionDocument.scala

```diff
@@ -0,0 +1,49 @@
+package edu.uci.ics.amber.engine.common.storage.partition
+
+import edu.uci.ics.amber.engine.common.storage.{FileDocument, VirtualDocument}
+
+import java.net.URI
+
+// PartitionDocument is a storage object that consists #numOfPartition physical files as its underlying data storage
+// Each underlying file's URI is in the format of {partitionDocumentURI}_{index}
+class PartitionDocument(val uri: URI, val numOfPartition: Int)
```

Comment:
I don't see an API to let multiple threads/workers to each write a partition. I guess I am looking for something like `partitionDocument.writePartition(partition: Int, content: String)`

Reply:
using current design, this call would look like: `partitionDocument.readItem(partitionIndex: Int).writeWithStream(content: InputStream)`